pkg/commands/os.go

```diff
@@ -70,16 +70,26 @@ func (c *OSCommand) DetectUnamePass(command string, ask func(string) string) err
 	errMessage, err := c.RunCommandWithOutputLive(command, func(word string) string {
 		ttyText = ttyText + " " + word
 
-		// prompt and patterns to check if the user needs to input a username / password
-		prompts := map[string]string{
-			"password": `Password\s*for\s*'.+':`,
-			"username": `Username\s*for\s*'.+':`,
+		type Prompt struct {
+			pattern   string
+			canAskFor bool
+		}
+		prompts := map[string]Prompt{
+			"password": Prompt{
+				pattern:   `Password\s*for\s*'.+':`,
+				canAskFor: true,
+			},
+			"username": Prompt{
+				pattern:   `Username\s*for\s*'.+':`,
+				canAskFor: true,
+			},
 		}
 
-		for prompt, pattern := range prompts {
-			if match, _ := regexp.MatchString(pattern, ttyText); match {
+		for askFor, propmt := range prompts {
```

Comment:
typo here, `propmt` should be `prompt`.

Also, I'm not sure why we need `canAskFor` here. We return as soon as we find a matching prompt, so it shouldn't matter what the value of canAskFor is

Reply:
I have added `canAskFor` because of the what if the commit message has the value `Password for ...` (a problem you pointed out earlier) if i don't include this lazygit will ask for a password or username and after that crach if the commit message includes `password for...`